convert_hf_to_gguf.py

```diff
@@ -262,9 +265,17 @@ def extra_f16_tensors(self, name: str, new_name: str, bid: int | None, n_dims: i
         return False
 
     def prepare_tensors(self):
+
+        uuidv5_sha1 = hashlib.sha1()
+        uuidv5_sha1.update(uuid.UUID('ef001206-dadc-5f6d-a15f-3359e577d4e5').bytes)
+
         max_name_len = max(len(s) for _, s in self.tensor_map.mapping.values()) + len(".weight,")
 
         for name, data_torch in self.get_tensors():
+
+            uuidv5_data_buffer: np.ndarray = data_torch.to(torch.float64).numpy()
+            uuidv5_sha1.update(uuidv5_data_buffer.tobytes('C'))
```

Comment:
```suggestion
            uuidv5_data_buffer: np.ndarray = data_torch.numpy()
            uuidv5_sha1.update(uuidv5_data_buffer.data.data)
```

Going to `float64` is not necessary. Hashing the bits directly should be fine.

Note that calculating the hash inline like this *still* reads all of the tensors twice because of lazy conversion. There isn't really a way around this except by keeping everything in memory (aka `--no-lazy`).

Reply:
For https://huggingface.co/Maykeye/TinyLLama-v0 which consistent of `torch.bfloat16` encoded tensors, I get something like

```
INFO:hf-to-gguf:gguf: loading model part 'model.safetensors'
Traceback (most recent call last):
  File "/home/mofosyne/gitextern/llama.cpp/convert_hf_to_gguf.py", line 3711, in <module>
    main()
  File "/home/mofosyne/gitextern/llama.cpp/convert_hf_to_gguf.py", line 3705, in main
    model_instance.write()
  File "/home/mofosyne/gitextern/llama.cpp/convert_hf_to_gguf.py", line 421, in write
    self.prepare_tensors()
  File "/home/mofosyne/gitextern/llama.cpp/convert_hf_to_gguf.py", line 1594, in prepare_tensors
    super().prepare_tensors()
  File "/home/mofosyne/gitextern/llama.cpp/convert_hf_to_gguf.py", line 276, in prepare_tensors
    uuidv5_data_buffer: np.ndarray = data_torch.numpy()
  File "/home/mofosyne/gitextern/llama.cpp/convert_hf_to_gguf.py", line 3533, in numpy
    dtype = self._dtype_map[self.dtype]
KeyError: torch.bfloat16
```

because there is no corresponding mapping in `_dtype_map` and also `torch.bfloat16` doesn't have a corresponding numpy type so can't easily map it across transparently. Had a similar issue before, hence why I was trying to convert to a fixed type in numpy first.

Reply:
In that case, maybe using something like `data_torch.view(torch.int8).numpy()` could help? Not converting, but re-intetpreting the data with an 8-bit type. The values themselves don't really need to be operated upon, but the hashing function needs the bits of the values.

But if you want to convert to a common type, then use `float32`, not `float64`. But know that even if the values are the same, the bits representing them will differ if their type is converted.